internal/restorer/fileswriter.go

```diff
@@ -60,6 +61,13 @@ func (w *filesWriter) writeToFile(path string, blob []byte, offset int64, create
 		bucket.files[path] = wr
 		bucket.users[path] = 1
 
+		if createSize >= 0 {
+			err := preallocateFile(wr, createSize)
+			if err != nil {
+				debug.Log("Failed to preallocate %v with size %v: %v", path, createSize, err)
+			}
```

Comment:
I intentionally did not return the error, as a preallocate call that fails for some reason could otherwise prevent restore from working.

There's at least one legitimate reason for preallocate to return an error: Not all filesystems or their implementations support preallocation which will yield a `syscall.ENOTSUP` error. Judging from the etcd implementation of their preallocate helpers in https://github.com/etcd-io/etcd/blob/master/pkg/fileutil/preallocate_darwin.go#L46 and https://github.com/etcd-io/etcd/blob/master/pkg/fileutil/preallocate_unix.go#L31  there's also a small assortment of further errnos when the call fails for some reason.

Reply:
Can you add this explanation (or abbreviated version of it) as a comment why preallocation errors are okay to ignore? And a note in release notes that preallocation can silently fail if not supported by filesystem.

Reply:
I've added the comment. For the note in the release notes I've reused the changelog entry for the new restorer implementation.